### ISSUE TITLE
OpenJDK 17 Ignite.Start, custom JvmDllPath

### ISSUE THREAD

**Issue by wasabii** [apache/ignite#10597]
Not sure what's up with this. I'm trying to launch Ignite from within a .NET program, running Apache.Ignite 2.14.0, against OpenJDK 17.

Upon Ignition.Start, I'm receiving:

```
JavaException: java.lang.ExceptionInInitializerError
	at org.apache.ignite.internal.processors.platform.memory.PlatformAbstractMemory.<clinit>(PlatformAbstractMemory.java:27)
	at org.apache.ignite.internal.processors.platform.PlatformAbstractBootstrap.init(PlatformAbstractBootstrap.java:54)
	at org.apache.ignite.internal.processors.platform.PlatformIgnition.start(PlatformIgnition.java:65)
Caused by: java.lang.RuntimeException: java.nio.DirectByteBuffer.address field is unavailable.
	at org.apache.ignite.internal.util.GridUnsafe$2.run(GridUnsafe.java:1570)
	at org.apache.ignite.internal.util.GridUnsafe$2.run(GridUnsafe.java:1557)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.apache.ignite.internal.util.GridUnsafe.bufferAddressOffset(GridUnsafe.java:1557)
	at org.apache.ignite.internal.util.GridUnsafe.<clinit>(GridUnsafe.java:109)
	... 3 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field long java.nio.Buffer.address accessible: module java.base does not "opens java.nio" to unnamed module @30946e09
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)
	at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(Unknown Source)
	at java.base/java.lang.reflect.Field.checkCanSetAccessible(Unknown Source)
	at java.base/java.lang.reflect.Field.setAccessible(Unknown Source)
	at org.apache.ignite.internal.util.GridUnsafe$2.run(GridUnsafe.java:1562)
	... 7 more

```

My guess is it's probably some of these missing VM options that are supposed to be passed, and mentioned in other sources. However, I'd expect the .NET library to set all that up automatically. Maybe it's not. Will let ya'll know.

**Comment by wasabii** [comment]
That moved it. The .NET library should probably be setting all this stuff up automatically.

**Comment by ptupitsyn** [comment]
> The .NET library should probably be setting all this stuff up automatically.

Yes, that was fixed in Ignite 2.15: https://issues.apache.org/jira/browse/IGNITE-18191
